src/parse.rs

```diff
@@ -1945,8 +1945,20 @@ mod tests {
     #[test]
     fn invalid_backref() {
         // only syntactic tests; see similar test in analyze module
-        fail(".\\12345678"); // unreasonably large number
```

Comment:
This was probably meant to be a number that is outside of e.g. usize range, right? We can change it to be a bigger number so that's actually being tested.